### ISSUE TITLE
How did you install mongodb on your Raspberry Pi 3?

### ISSUE THREAD

**Issue by HeatfanJohn** [Syps/alexa-airplane-spotter#2]
How did you install mongodb on your Raspberry Pi 3?

The Mongodb site only has packages for 64-bit ARM for Ubuntu?

Did you compile from source?

**Comment by Syps** [comment]
I installed from the Rasbian repo (`apt-get install mongodb`). I believe the MongoDB version there is held back to an earlier version so as to maintain compatibility w/ the Pi's 32-bit OS.

**Comment by HeatfanJohn** [comment]
Thanks!  I didn't think to look there!

I also found this http://andyfelong.com/2016/01/mongodb-3-0-9-binaries-for-raspberry-pi-2-jessie/ and this http://andyfelong.com/2017/03/mongodb-3-0-14-binaries-for-raspberry-pi-3/

I manually installed the 3.0.14 version.

What version is in the Raspbian repo?

**Comment by HeatfanJohn** [comment]
FYI, I used the following commands to import aircraft_db.csv and airports_tz.json into the MongoDB:

     mongoimport -d AircraftData -c Registration --type csv --file aircraft_db.csv --headerline
     mongoimport -d AircraftData -c AirportTZ < airports_tz.json

**Comment by HeatfanJohn** [comment]
Additional information regarding Mongo.  I had to transform the `airports_tz.json` to be able to load the data into Mongo for use by the application.  The original copy of the file appears to only have one Mongo document containing all of the Airport codes and associated timezones.  I wasn't able to get the Mongo `findOne` method to located a timezone given an airport code.

I used `vi` to edit the file to have one individual Mongo document for each airport code.  I was able to load the transformed json file into Mongo using:

    mongoimport -d AircraftData -c AirportTZ < airports_tz.json
```
> db.AirportTZ.findOne( {'code': 'FLL'} )
{
        "_id" : ObjectId("59714e9a357e6c6ca81c0531"),
        "timezone" : "America/New_York",
        "code" : "FLL",
        "offset" : {
                "dst" : -4,
                "gmt" : -5
        }
}
>
```
I transformed the data from this:
```
{"MLC": {"timezone": "America/Chicago", "code": "MLC", "offset": {"dst": -5, "gmt": -6}},
 "NDP": {"timezone": "America/Chicago", "code": "NDP", "offset": {"dst": null, "gmt": null}},
 "NDS": {"timezone": "Australia/Perth", "code": "NDS", "offset": {"dst": 8, "gmt": 8}},
 "SPX": {"timezone": "America/Chicago", "code": "SPX", "offset": {"dst": -5, "gmt": -6}},
```
To this:
```
{"timezone": "America/Chicago", "code": "MLC", "offset": {"dst": -5, "gmt": -6}}
{"timezone": "America/Chicago", "code": "NDP", "offset": {"dst": null, "gmt": null}}
{"timezone": "Australia/Perth", "code": "NDS", "offset": {"dst": 8, "gmt": 8}}
{"timezone": "America/Chicago", "code": "SPX", "offset": {"dst": -5, "gmt": -6}} 
```
I added carriage returns above to make the data more legible.

I want to thank you for creating such a great and useful application.  Reading your code has helped me learn new ways to write python code.

**Comment by Syps** [comment]
Thanks a lot for the insight on using `mongoimport` as well as the feedback! As you pointed out in your last comment, using this tool instead of running `load_tz_db.py` simply requires reformatting `airports_tz.json`, but it does seem like a cleaner option. I will keep this in mind going forward :)

As for the MongoDB version, running `apt-cache show mongodb` on the Pi shows a current version of `1:2.4.10-5+deb8u1`, so using the binaries you link to would definitely provide a more up-to-date MongoDB, though for the purposes of this project the Rasbian version should do the job.

Closing this issue now, but do let me know if any other issues or insights arise.

**Comment by HeatfanJohn** [comment]
Oh, sorry, I didn't realize that you had two Python programs to load the Mongo data.

FYI, I got the Alexa skill working.  I changed `shouldEndSession` to `true` to have Alexa end the session after saying the results otherwise any conversation heard by Alexa after saying the results caused her make another lambda call.

I also needed to create a deployment zip for the lambda function because `requests` isn't included in the base Python for AWS Lambda functions.

I used `pip install requests -t .` to locally install requests in my deployment zip.
